ostd/src/mm/page_table/cursor.rs

```diff
@@ -565,6 +568,19 @@ where
             {
                 if cur_pte.is_present() && !cur_pte.is_last(self.0.level) {
                     self.0.level_down();
+
+                    // We have got down a level. If there's no mapped PTEs in
+                    // the current node, we can go back and skip to save time.
+                    if self.0.guards[(self.0.level - 1) as usize]
+                        .as_ref()
```

Comment:
I understand that the purpose of this branch should be to level up when the upper node has no mapped child nodes, in order to avoid subsequent loops, right? But I feel that this branch should be placed after unsetting the current node (line 594). Am I missing something?

Reply:
This is to circumvent the "page table recycling" problem. If the current node is unset the page would be recycled for other usages. We should not recycle any page table pages due to the following reasons, which is stated in the documentation of `page_walk`:

https://github.com/asterinas/asterinas/blob/5aa28eae7e14594bbe68827114443b31002bf742/ostd/src/mm/page_table/mod.rs#L283-L293

In the future, we may design a recycle policy after inter-process TLB eviction is implemented.